cockpit/src/subscriptions-client.js

```diff
@@ -124,161 +92,104 @@ function getSubscriptionDetails() {
+            (?::(?=[0-9])([0-9]+))? matches the port if it exists
+            (?:(/.+)) matches the rest for the path
+        */
+        pattern = new RegExp('^(?:https?://)?([^/:]+)(?::(?=[0-9])([0-9]+))?(?:(/.+))?$');
+        match = pattern.exec(subscriptionDetails.url); // TODO handle failure
+        options.host = {
+            t: 's',
+            v: match[1],
+        };
+        options.port = {
+            t: 's',
+            v: match[2],
+        };
+        options.handler = {
+            t: 's',
+            v: match[3],
+        };
     }
 
+    var connection_options = {};
     // proxy is optional
     if (subscriptionDetails.proxy) {
-        if (!subscriptionDetails.proxyServer)
-            subscriptionDetails.proxyServer = '';
-        if (!subscriptionDetails.proxyUser)
-            subscriptionDetails.proxyUser = '';
-        if (!subscriptionDetails.proxyPass)
-            subscriptionDetails.proxyPass = '';
-        args.push('--proxy', subscriptionDetails.proxyServer,
-                  '--proxyuser', subscriptionDetails.proxyUser,
-                  '--proxypass', subscriptionDetails.proxyPass);
+        connection_options.proxy_hostname = {
+            t: 's',
+            v: subscriptionDetails.proxyServer,
+        };
+        connection_options.proxy_user = {
+            t: 's',
+            v: subscriptionDetails.proxyUser,
+        }
+        connection_options.proxy_password = {
+            t: 's',
+            v: subscriptionDetails.proxyPass,
+        }
     }
 
-    // only pass org info if user provided it
-    if (subscriptionDetails.org)
-        args.push('--org', subscriptionDetails.org);
-
-    /* TODO DBus API doesn't deliver what we need, so we call subscription manager
-     * without translations and parse the output
-     * https://bugzilla.redhat.com/show_bug.cgi?id=1304056
-     */
-    var process = cockpit.spawn(args, {
-        directory: '/',
-        superuser: "require",
-        environ: ['LC_ALL=C'],
-        err: "out"
-    });
-
-    var promise;
-    var buffer = '';
-    process
-        .input('')
-        .stream(function(text) {
-            buffer += text;
-        })
-        .done(function(output) {
-            dfd.resolve();
-        })
-        .fail(function(ex) {
-            if (ex.problem === "cancelled") {
-                dfd.reject(ex);
-                return;
+    registerServer.Start()
+        .then(function(socket) {
+            console.debug('Opening private bus interface at ' + socket);
+            var private_interface = cockpit.dbus(null, {bus: 'none', address: socket, superuser: 'require'});
+            var registerService = private_interface.proxy('com.redhat.RHSM1.Register', '/com/redhat/RHSM1/Register');
+            if (subscriptionDetails.activationKeys) {
+                return registerService.call('RegisterWithActivationKeys', [subscriptionDetails.org, subscriptionDetails.activationKeys, options, connection_options]);
```

Comment:
You can replace proxy.call() with the actual name of the method as long as the method name is UpperCase.

```javascript
return registerService.RegisterWithActivationKeys([subscriptionDetails.org, subscriptionDetails.activationKeys, options, connection_options]);
```

You can do this in some of the other places you have call as well.